src/java/org/apache/cassandra/thrift/CustomTThreadPoolServer.java

```diff
@@ -229,6 +242,9 @@ public void run()
                 if (socket != null)
                     ThriftSessionManager.instance.connectionComplete(socket);
 
+                InetSocketAddress clientAddressAndIp = (InetSocketAddress) ((TCustomSocket) client_).getSocket().getRemoteSocketAddress();
+                InetAddress clientAddress = clientAddressAndIp.getAddress();
+                connections.computeIfPresent(clientAddress, (_ip, count) -> count - 1);
```

Comment:
I think this will leak memory unless we remove values once they reach zero.
Also worth using a wrapper around the count value otherwise we'll constantly box and unbox integers.